File: LeetCode/0190.reverse-bits/reverse-bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint64_t flag = INT_MAX + 1L;
        uint32_t res = 0,x = 1;
        while (flag){
            if (flag & n){
                res += x;
            }
            x <<= 1;
            flag >>= 1;

        }
        return res;
    }
};

File: LeetCode/0190.reverse-bits/reverse-bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        long flag = Integer.MAX_VALUE + 1L;
        int res = 0,x = 1;
        while (flag != 0){
            if ((n & flag) != 0){
                res += x;
            }
            x <<= 1;;
            flag >>= 1;
        }
        return res;
    }
}